### PR TITLE
Pin PyParsing version to be compatible with pip 20

### PR DESCRIPTION
**Pull request by dyt811**
#### GitHub

- [X] I've given this PR a concise, self-descriptive, and meaningful title
- [X] I've linked relevant issues in the PR body
- [X] I've applied [the relevant labels](https://www.neuro.polymtl.ca/software/contributing#pr_labels) to this PR
- [ ] I've assigned a reviewer

#### PR contents

- [X] I've consulted [ivadomed's internal developer documentation](https://github.com/ivadomed/ivadomed/wiki) to ensure my contribution is in line with any relevant design decisions
- [ ] I've added [relevant tests](https://github.com/ivadomed/ivadomed/wiki/tests) for my contribution
- [ ] I've updated the [relevant documentation](https://github.com/ivadomed/ivadomed/wiki/documentation) for my changes, including argparse descriptions, docstrings, and ReadTheDocs tutorial pages

## Description

When testing `ivadomed` on Romane and other super computing clusters with pip 20, we ran into error that prevent scripts from being launched. See details at https://github.com/neuropoly/computers/issues/218
TLDR: pyparsing version conflict cause issue with registering binary commands such as `ivaodmed_download`

Note that this issue is not reproducible in CI which typically has updated PIP version. It only shows up when PIP version is at 20

## Linked issues

Resovles https://github.com/neuropoly/computers/issues/218

### CODE REVIEW

**Comment by kousu** [comment]
Here's my description, summarized from https://github.com/neuropoly/computers/issues/218:

Installing ivadomed **in a virtualenv** on **Ubuntu 20.04** fails with:

> ERROR: packaging 21.2 has requirement pyparsing<3,>=2.0.2, but you'll have pyparsing 3.0.5 which is incompatible.

<details>

```
u108545@romane:~$ virtualenv venv-ivadomed
created virtual environment CPython3.8.10.final.0-64 in 147ms
  creator CPython3Posix(dest=/home/GRAMES.POLYMTL.CA/u108545/venv-ivadomed, clear=False, global=False)
  seeder FromAppData(download=False, pip=latest, setuptools=latest, wheel=latest, pkg_resources=latest, via=copy, app_data_dir=/home/GRAMES.POLYMTL.CA/u108545/.local/share/virtualenv/seed-app-data/v1.0.1.debian.1)
  activators BashActivator,CShellActivator,FishActivator,PowerShellActivator,PythonActivator,XonshActivator
u108545@romane:~$ source venv-ivadomed/bin/activate
(venv-ivadomed) u108545@romane:~$ git clone https://github.com/ivadomed/ivadomed.git
Cloning into 'ivadomed'...
cd ivadomed
remote: Enumerating objects: 25607, done.
remote: Counting objects: 100% (3269/3269), done.
remote: Compressing objects: 100% (1459/1459), done.
remote: Total 25607 (delta 2427), reused 2505 (delta 1804), pack-reused 22338
Receiving objects: 100% (25607/25607), 133.13 MiB | 37.32 MiB/s, done.
Resolving deltas: 100% (18749/18749), done.
(venv-ivadomed) u108545@romane:~$ cd ivadomed
(venv-ivadomed) u108545@romane:~/ivadomed$ pip install -r requirements_gpu.txt
Ignoring h5py: markers 'python_version == "3.9"' don't match your environment
Looking in links: https://download.pytorch.org/whl/torch_stable.html
Collecting csv-diff>=1.0
  Using cached csv_diff-1.1-py3-none-any.whl (12 kB)
Collecting h5py==2.10.0
  Using cached h5py-2.10.0-cp38-cp38-manylinux1_x86_64.whl (2.9 MB)
Collecting loguru~=0.5
  Using cached loguru-0.5.3-py3-none-any.whl (57 kB)
Collecting joblib~=1.0
  Using cached joblib-1.1.0-py2.py3-none-any.whl (306 kB)
Collecting matplotlib>=3.3.0
  Using cached matplotlib-3.4.3-cp38-cp38-manylinux1_x86_64.whl (10.3 MB)
Collecting nibabel~=3.2
  Using cached nibabel-3.2.1-py3-none-any.whl (3.3 MB)
Collecting onnxruntime==1.7.0
  Using cached onnxruntime-1.7.0-cp38-cp38-manylinux2014_x86_64.whl (4.1 MB)
Collecting pandas~=1.1
  Using cached pandas-1.3.4-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (11.5 MB)
Collecting pillow>=7.0.0
  Downloading Pillow-8.4.0-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.1 MB)
     |████████████████████████████████| 3.1 MB 34.1 MB/s 
Collecting pybids>=0.12.4
  Using cached pybids-0.13.2-py3-none-any.whl (3.2 MB)
Collecting scikit-learn>=0.20.3
  Using cached scikit_learn-1.0.1-cp38-cp38-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (25.9 MB)
Collecting scikit-image~=0.17
  Using cached scikit_image-0.18.3-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.whl (30.2 MB)
Collecting seaborn~=0.11
  Using cached seaborn-0.11.2-py3-none-any.whl (292 kB)
Collecting tensorboard~=2.4
  Downloading tensorboard-2.7.0-py3-none-any.whl (5.8 MB)
     |████████████████████████████████| 5.8 MB 8.6 MB/s 
Collecting tqdm>=4.30
  Using cached tqdm-4.62.3-py2.py3-none-any.whl (76 kB)
Collecting scipy
  Downloading scipy-1.7.2-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (39.3 MB)
     |████████████████████████████████| 39.3 MB 69.0 MB/s 
Collecting torch==1.8.1+cu111
  Downloading https://download.pytorch.org/whl/cu111/torch-1.8.1%2Bcu111-cp38-cp38-linux_x86_64.whl (1982.2 MB)
     |████████████████████████████████| 1982.2 MB 13 kB/s 
Collecting torchvision==0.9.1+cu111
  Downloading https://download.pytorch.org/whl/cu111/torchvision-0.9.1%2Bcu111-cp38-cp38-linux_x86_64.whl (17.6 MB)
     |████████████████████████████████| 17.6 MB 57.6 MB/s 
Collecting dictdiffer
  Using cached dictdiffer-0.9.0-py2.py3-none-any.whl (16 kB)
Collecting click
  Using cached click-8.0.3-py3-none-any.whl (97 kB)
Collecting six
  Downloading six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting numpy>=1.7
  Downloading numpy-1.21.4-cp38-cp38-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (15.7 MB)
     |████████████████████████████████| 15.7 MB 76.3 MB/s 
Collecting cycler>=0.10
  Using cached cycler-0.11.0-py3-none-any.whl (6.4 kB)
Collecting python-dateutil>=2.7
  Using cached python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
Collecting pyparsing>=2.2.1
  Downloading pyparsing-3.0.5-py3-none-any.whl (97 kB)
     |████████████████████████████████| 97 kB 17.1 MB/s 
Collecting kiwisolver>=1.0.1
  Using cached kiwisolver-1.3.2-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.whl (1.2 MB)
Collecting packaging>=14.3
  Downloading packaging-21.2-py3-none-any.whl (40 kB)
     |████████████████████████████████| 40 kB 11.6 MB/s 
Collecting protobuf
  Downloading protobuf-3.19.1-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.1 MB)
     |████████████████████████████████| 1.1 MB 81.2 MB/s 
Collecting pytz>=2017.3
  Using cached pytz-2021.3-py2.py3-none-any.whl (503 kB)
Collecting num2words
  Using cached num2words-0.5.10-py3-none-any.whl (101 kB)
Collecting sqlalchemy<1.4.0.dev0
  Using cached SQLAlchemy-1.3.24-cp38-cp38-manylinux2010_x86_64.whl (1.3 MB)
Collecting bids-validator
  Using cached bids_validator-1.8.4-py2.py3-none-any.whl (19 kB)
Collecting patsy
  Using cached patsy-0.5.2-py2.py3-none-any.whl (233 kB)
Collecting threadpoolctl>=2.0.0
  Using cached threadpoolctl-3.0.0-py3-none-any.whl (14 kB)
Collecting networkx>=2.0
  Using cached networkx-2.6.3-py3-none-any.whl (1.9 MB)
Collecting PyWavelets>=1.1.1
  Using cached PyWavelets-1.1.1-cp38-cp38-manylinux1_x86_64.whl (4.4 MB)
Collecting tifffile>=2019.7.26
  Using cached tifffile-2021.11.2-py3-none-any.whl (178 kB)
Collecting imageio>=2.3.0
  Using cached imageio-2.10.3-py3-none-any.whl (3.3 MB)
Requirement already satisfied: wheel>=0.26 in /home/GRAMES.POLYMTL.CA/u108545/venv-ivadomed/lib/python3.8/site-packages (from tensorboard~=2.4->-r requirements_common.txt (line 15)) (0.34.2)
Collecting absl-py>=0.4
  Downloading absl_py-0.15.0-py3-none-any.whl (132 kB)
     |████████████████████████████████| 132 kB 91.4 MB/s 
Collecting grpcio>=1.24.3
  Downloading grpcio-1.41.1-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.9 MB)
     |████████████████████████████████| 3.9 MB 83.0 MB/s 
Collecting tensorboard-plugin-wit>=1.6.0
  Downloading tensorboard_plugin_wit-1.8.0-py3-none-any.whl (781 kB)
     |████████████████████████████████| 781 kB 72.9 MB/s 
Collecting google-auth-oauthlib<0.5,>=0.4.1
  Downloading google_auth_oauthlib-0.4.6-py2.py3-none-any.whl (18 kB)
Collecting google-auth<3,>=1.6.3
  Downloading google_auth-2.3.3-py2.py3-none-any.whl (155 kB)
     |████████████████████████████████| 155 kB 138.7 MB/s 
Collecting werkzeug>=0.11.15
  Downloading Werkzeug-2.0.2-py3-none-any.whl (288 kB)
     |████████████████████████████████| 288 kB 49.5 MB/s 
Collecting requests<3,>=2.21.0
  Downloading requests-2.26.0-py2.py3-none-any.whl (62 kB)
     |████████████████████████████████| 62 kB 2.8 MB/s 
Collecting markdown>=2.6.8
  Downloading Markdown-3.3.4-py3-none-any.whl (97 kB)
     |████████████████████████████████| 97 kB 18.9 MB/s 
Requirement already satisfied: setuptools>=41.0.0 in /home/GRAMES.POLYMTL.CA/u108545/venv-ivadomed/lib/python3.8/site-packages (from tensorboard~=2.4->-r requirements_common.txt (line 15)) (44.0.0)
Collecting tensorboard-data-server<0.7.0,>=0.6.0
  Downloading tensorboard_data_server-0.6.1-py3-none-manylinux2010_x86_64.whl (4.9 MB)
     |████████████████████████████████| 4.9 MB 91.9 MB/s 
Collecting typing-extensions
  Downloading typing_extensions-3.10.0.2-py3-none-any.whl (26 kB)
Processing /home/GRAMES.POLYMTL.CA/u108545/.cache/pip/wheels/56/ea/58/ead137b087d9e326852a851351d1debf4ada529b6ac0ec4e8c/docopt-0.6.2-py2.py3-none-any.whl
Collecting requests-oauthlib>=0.7.0
  Downloading requests_oauthlib-1.3.0-py2.py3-none-any.whl (23 kB)
Collecting rsa<5,>=3.1.4; python_version >= "3.6"
  Downloading rsa-4.7.2-py3-none-any.whl (34 kB)
Collecting pyasn1-modules>=0.2.1
  Downloading pyasn1_modules-0.2.8-py2.py3-none-any.whl (155 kB)
     |████████████████████████████████| 155 kB 53.5 MB/s 
Collecting cachetools<5.0,>=2.0.0
  Downloading cachetools-4.2.4-py3-none-any.whl (10 kB)
Collecting urllib3<1.27,>=1.21.1
  Downloading urllib3-1.26.7-py2.py3-none-any.whl (138 kB)
     |████████████████████████████████| 138 kB 83.4 MB/s 
Collecting charset-normalizer~=2.0.0; python_version >= "3"
  Downloading charset_normalizer-2.0.7-py3-none-any.whl (38 kB)
Collecting idna<4,>=2.5; python_version >= "3"
  Downloading idna-3.3-py3-none-any.whl (61 kB)
     |████████████████████████████████| 61 kB 13.8 MB/s 
Collecting certifi>=2017.4.17
  Downloading certifi-2021.10.8-py2.py3-none-any.whl (149 kB)
     |████████████████████████████████| 149 kB 68.0 MB/s 
Collecting oauthlib>=3.0.0
  Downloading oauthlib-3.1.1-py2.py3-none-any.whl (146 kB)
     |████████████████████████████████| 146 kB 87.4 MB/s 
Collecting pyasn1>=0.1.3
  Downloading pyasn1-0.4.8-py2.py3-none-any.whl (77 kB)
     |████████████████████████████████| 77 kB 14.0 MB/s 
ERROR: packaging 21.2 has requirement pyparsing<3,>=2.0.2, but you'll have pyparsing 3.0.5 which is incompatible.
Installing collected packages: dictdiffer, click, csv-diff, six, numpy, h5py, loguru, joblib, cycler, python-dateutil, pillow, pyparsing, kiwisolver, matplotlib, packaging, nibabel, protobuf, onnxruntime, pytz, pandas, docopt, num2words, sqlalchemy, bids-validator, scipy, patsy, pybids, threadpoolctl, scikit-learn, networkx, PyWavelets, tifffile, imageio, scikit-image, seaborn, absl-py, grpcio, tensorboard-plugin-wit, oauthlib, urllib3, charset-normalizer, idna, certifi, requests, requests-oauthlib, pyasn1, rsa, pyasn1-modules, cachetools, google-auth, google-auth-oauthlib, werkzeug, markdown, tensorboard-data-server, tensorboard, tqdm, typing-extensions, torch, torchvision
Successfully installed PyWavelets-1.1.1 absl-py-0.15.0 bids-validator-1.8.4 cachetools-4.2.4 certifi-2021.10.8 charset-normalizer-2.0.7 click-8.0.3 csv-diff-1.1 cycler-0.11.0 dictdiffer-0.9.0 docopt-0.6.2 google-auth-2.3.3 google-auth-oauthlib-0.4.6 grpcio-1.41.1 h5py-2.10.0 idna-3.3 imageio-2.10.3 joblib-1.1.0 kiwisolver-1.3.2 loguru-0.5.3 markdown-3.3.4 matplotlib-3.4.3 networkx-2.6.3 nibabel-3.2.1 num2words-0.5.10 numpy-1.21.4 oauthlib-3.1.1 onnxruntime-1.7.0 packaging-21.2 pandas-1.3.4 patsy-0.5.2 pillow-8.4.0 protobuf-3.19.1 pyasn1-0.4.8 pyasn1-modules-0.2.8 pybids-0.13.2 pyparsing-3.0.5 python-dateutil-2.8.2 pytz-2021.3 requests-2.26.0 requests-oauthlib-1.3.0 rsa-4.7.2 scikit-image-0.18.3 scikit-learn-1.0.1 scipy-1.7.2 seaborn-0.11.2 six-1.16.0 sqlalchemy-1.3.24 tensorboard-2.7.0 tensorboard-data-server-0.6.1 tensorboard-plugin-wit-1.8.0 threadpoolctl-3.0.0 tifffile-2021.11.2 torch-1.8.1+cu111 torchvision-0.9.1+cu111 tqdm-4.62.3 typing-extensions-3.10.0.2 urllib3-1.26.7 werkzeug-2.0.2
```

</details>

The problem is that Ubuntu 20.04 comes with a buggy pip. This adds some **temporary** glue to handle this, until the new Ubuntu LTS comes out when we can assume most people will safely be using the [bug-fixed `pip`](https://pip.pypa.io/en/stable/user_guide/#changes-to-the-pip-dependency-resolver-in-20-3-2020).

An equally good fix is to run `virtualenv ivadomed-venv && source ivadomed-venv/bin/activate && pip install --upgrade pip && pip install ivadomed`. We could update [the docs](https://ivadomed.org/installation.html#approach-2-venv) to recommend this too, but in this case I think the hackier solution is nicer because it has wider world compatibility.

Oddly, installing with `pip install --user` instead of `virtualenv ivadomed-venv && source ivadomed-venv/bin/activate && pip install ivadomed` will probably work, since `pyparsing` and `packaging` are pre-installed on most Ubuntu 20.04 systems, as system packages, *in older, compatible versions*. It's just that `virtualenv` **on Ubuntu** skips OS python packages (because Ubuntu puts those into `/usr/local/lib/python*/dist-packages`, which isn't on `virtualenv`'s default `sys.path`), triggering a fresh download, of the latest of both, causing the conflict.

File: requirements_common.txt [hunk 1]
```diff
@@ -1,3 +1,5 @@
+# workarounds for pip 20.0.2's buggy dependency resolver, which doesn't account for cross-package constraints
+pyparsing<3,>=2.0.2
```
Comment (kousu) [inline requirements_common.txt:2]:
```suggestion
# workarounds for pip 20.0.2's buggy dependency resolver
# which doesn't account for cross-package constraints: https://pip.pypa.io/en/stable/user_guide/#changes-to-the-pip-dependency-resolver-in-20-3-2020
# matplotlib pulls in the latest pyparsing, nibabel pulls in the latest packaging, and the two latests are in conflict.
# > ERROR: packaging 21.2 has requirement pyparsing<3,>=2.0.2, but you'll have pyparsing 3.0.5 which is incompatible.
# This forces the older pip, if the user has the older pip, to behave itself.
# This is to specially support users on Ubuntu 20.04 LTS; when Ubuntu 22.04 LTS comes out, this can be removed.
pyparsing<3,>=2.0.2
```

Comment (kousu) [reply requirements_common.txt:2]: How about this?

**Review by kousu — CHANGES_REQUESTED** [review]
